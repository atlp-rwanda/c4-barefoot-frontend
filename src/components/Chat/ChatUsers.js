import React from 'react';
import { TextField, Typography, CssBaseline, AppBar, makeStyles, Toolbar, IconButton, Fab, ListItem, ListItemAvatar, ListItemText, ListSubheader, Avatar, Paper, List } from "@material-ui/core";
import { Person } from "@material-ui/icons";
import { connect } from 'react-redux';
import { fetchUsersChat } from '../../redux/actions/ChatAction';
import UsersList from './UsersList';

class ChatUsers extends React.Component {
    UNSAFE_componentWillMount() {
        this.props.fetchUsersChat();
    }
    render () {
        const users = this.props.users;
        return (
            <div>
                <CssBaseline/>
                <Paper position="fixed">
                    <TextField
                        id="outlined-basic"
                        variant="outlined"
                        label="Find Users" 
                        variant="filled"
                        style={{boxSizing: "border-box",width: '100%', padding: '10px', borderRadius: '50px'}}
                    ><Person/></TextField>
                    <UsersList users={users}/>
                    <Typography>VISITORS</Typography>
                </Paper>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    users: state.chat.users
})

export default connect(mapStateToProps, { fetchUsersChat })(ChatUsers)