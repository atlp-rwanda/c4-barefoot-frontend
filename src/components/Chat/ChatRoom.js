import React from 'react';
import { Grid, AppBar, Toolbar, TextField } from "@material-ui/core";
import ChatUsers from './ChatUsers';
import ChatMessages from './ChatMessages';
import { useStyles } from './ChatStyles';

const ChatRoom = () => {
    const classes = useStyles();
    return (
        <Grid container spacing={1}>
            <Grid item xs={12} sm={4} md={2}>
                <ChatUsers/>
            </Grid>
            <Grid item xs={12} sm={7} md={10}>
                <ChatMessages/>
            </Grid> 
        </Grid>
    )
}

export default ChatRoom;