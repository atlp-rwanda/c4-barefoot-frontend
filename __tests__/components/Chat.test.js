import React from 'react';
import ChatRoom from '../../src/components/Chat/ChatRoom';
import { shallow, mount } from 'enzyme';
import { expect } from 'chai';
import { Grid } from '@material-ui/core'
import ChatUsers from '../../src/components/Chat/ChatUsers';

describe('<ChatRoom /> tests', () => {
    it('Should render ChatRoom component without any errors', () => {
        const wrapper = shallow(<ChatRoom ><Grid/></ChatRoom>);
        expect(wrapper.find(<Grid></Grid>)).to.have.lengthOf(0);
    })
})