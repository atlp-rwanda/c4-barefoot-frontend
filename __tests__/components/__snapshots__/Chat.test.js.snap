// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatRoom /> should be rendered without errors 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  spacing={1}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={2}
    sm={4}
    xs={12}
  >
    <Connect(ChatUsers) />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
    md={10}
    sm={7}
    xs={12}
  >
    <Connect(ChatMessages) />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
